src/plugins/commander.go

```diff
@@ -163,11 +163,14 @@ func (c *Commander) agentRegistered(cmd *proto.Command) {
 				}
 			}
 
-			for index, feature := range agtCfg.Details.Features {
-				agtCfg.Details.Features[index] = strings.Replace(feature, "features_", "", 1)
+			if agtCfg.Details != nil && agtCfg.Details.Features != nil {
```

Comment:
nit: it's safe to `range` over a nil slice ([demo](https://go.dev/play/p/dADYKu93P57)), we can omit one of these checks

```suggestion
			if agtCfg.Details != nil {
```